### ISSUE TITLE
Does not start under MacOS Big Sur

### ISSUE THREAD

**Issue by jgoldhammer** [smittytone/mnu#1]
I have installed your app in MacOS Big Sur via homebrew.

![image](https://user-images.githubusercontent.com/3872101/110630452-a6071100-81a5-11eb-94c1-521e73db8c9f.png)

Starting the app does not bring up any window or menubar item.
Using the system logs I see following error:

`default	13:38:35.250080+0100	MNU	Could not cast value of type '__NSCFBoolean' (0x7fff8838bb98) to 'NSString' (0x7fff883bf060).
`

System Console Output filtered for MNU

```
Default	13:38:34.943675+0100	runningboardd	Launch request for app<application.com.bps.MNU.366350163.366350169(1875722432)> is using uid 1875722432 (divined from auid 1875722432 euid 1875722432)
default	13:38:34.943778+0100	runningboardd	Executing launch request for app<application.com.bps.MNU.366350163.366350169(1875722432)> (LS launch com.bps.MNU)
default	13:38:34.943833+0100	runningboardd	Creating and launching job for: app<application.com.bps.MNU.366350163.366350169(1875722432)>
default	13:38:34.943870+0100	runningboardd	_mutateContextIfNeeded called for com.bps.MNU
default	13:38:34.951508+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] This process will not be managed.
default	13:38:34.951543+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] HOME is <private>
default	13:38:34.951567+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] TMPDIR is <private>
default	13:38:34.951602+0100	runningboardd	Now tracking process: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:34.951673+0100	runningboardd	Using default underlying assertion for app: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:34.951714+0100	runningboardd	Calculated state for app<application.com.bps.MNU.366350163.366350169(1875722432)>: running-suspended (role: None)
default	13:38:34.951781+0100	runningboardd	Acquiring assertion targeting [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] from originator [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] with description <RBSAssertionDescriptor| "RB Underlying Assertion" ID:258-258-10518 target:63335 attributes:[
	<RBSDomainAttribute| domain:"com.apple.underlying" name:"defaultUnderlyingAppAssertion" sourceEnvironment:"(null)">,
	<RBSAcquisitionCompletionAttribute| policy:AfterApplication>
	]>
default	13:38:34.951967+0100	runningboardd	Assertion 258-258-10518 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) will be created as active
default	13:38:34.952305+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] reported to RB as running
default	13:38:34.952608+0100	runningboardd	Calculated state for app<application.com.bps.MNU.366350163.366350169(1875722432)>: running-suspended (role: None)
default	13:38:34.952620+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring jetsam update because this process is not memory-managed
default	13:38:34.952677+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring suspend because this process is not lifecycle managed
default	13:38:34.952831+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Set darwin role to: None
default	13:38:34.952915+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring GPU update because this process is not GPU managed
default	13:38:34.953097+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Skipping AppNap state - not lifecycle managed
default	13:38:34.953275+0100	runningboardd	Finished acquiring assertion 258-258-10518 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335])
default	13:38:34.953319+0100	runningboardd	Successfully acquired underlying assertion for [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:34.956253+0100	runningboardd	Acquiring assertion targeting [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] from originator [daemon<com.apple.coreservices.launchservicesd>:116] with description <RBSAssertionDescriptor| "uielement:63335" ID:258-116-10519 target:63335 attributes:[
	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"RoleUserInteractive" sourceEnvironment:"(null)">
	]>
default	13:38:34.956367+0100	runningboardd	Assertion 258-116-10519 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) will be created as active
default	13:38:34.956815+0100	runningboardd	Calculated state for app<application.com.bps.MNU.366350163.366350169(1875722432)>: running-active (role: UserInteractive)
default	13:38:34.956835+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring jetsam update because this process is not memory-managed
default	13:38:34.956844+0100	runningboardd	Acquiring assertion targeting [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] from originator [daemon<com.apple.coreservices.launchservicesd>:116] with description <RBSAssertionDescriptor| "uielement:63335" ID:258-116-10520 target:63335 attributes:[
	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"RoleUserInteractive" sourceEnvironment:"(null)">
	]>
default	13:38:34.956903+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring suspend because this process is not lifecycle managed
default	13:38:34.957115+0100	runningboardd	Assertion 258-116-10520 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) will be created as active
default	13:38:34.957133+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Set darwin role to: UserInteractive
default	13:38:34.957710+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Ignoring GPU update because this process is not GPU managed
default	13:38:34.958657+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] Skipping AppNap state - not lifecycle managed
default	13:38:34.958739+0100	runningboardd	Finished acquiring assertion 258-116-10519 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335])
default	13:38:34.958838+0100	runningboardd	Invalidating assertion 258-116-10519 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) from originator [daemon<com.apple.coreservices.launchservicesd>:116]
default	13:38:34.959028+0100	runningboardd	Finished acquiring assertion 258-116-10520 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335])
default	13:38:34.959652+0100	runningboardd	Acquiring assertion targeting [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] from originator [daemon<com.apple.coreservices.launchservicesd>:116] with description <RBSAssertionDescriptor| "uielement:63335" ID:258-116-10521 target:63335 attributes:[
	<RBSDomainAttribute| domain:"com.apple.launchservicesd" name:"RoleUserInteractive" sourceEnvironment:"(null)">
	]>
default	13:38:34.959908+0100	runningboardd	Assertion 258-116-10521 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) will be created as active
default	13:38:34.961071+0100	runningboardd	Invalidating assertion 258-116-10520 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]) from originator [daemon<com.apple.coreservices.launchservicesd>:116]
default	13:38:34.962093+0100	runningboardd	Finished acquiring assertion 258-116-10521 (target:[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335])
default	13:38:34.963376+0100	loginwindow	-[ApplicationManager checkInAppContext:eventData:] | ApplicationManager: Checked in app : MNU.app
default	13:38:34.984045+0100	MNU	CHECKIN: pid=63335
default	13:38:34.997169+0100	launchservicesd	CHECKIN:0x0-0x1d41d4 63335 com.bps.MNU
default	13:38:34.997754+0100	MNU	CHECKEDIN: pid=63335 asn=0x0-0x1d41d4 foreground=0
default	13:38:34.998087+0100	loginwindow	-[ApplicationManager checkInAppContext:eventData:] | ApplicationManager: Checked in app : MNU
default	13:38:35.005424+0100	tccd	AUTHREQ_ATTRIBUTION: msgID=63335.1, attribution={requesting={identifier=com.bps.MNU, pid=63335, auid=1875722432, euid=1875722432, binary_path=/Applications/MNU.app/Contents/MacOS/MNU}, },
default	13:38:35.013643+0100	tccd	AUTHREQ_SUBJECT: msgID=63335.1, subject=com.bps.MNU,
default	13:38:35.016829+0100	tccd	AUTHREQ_ATTRIBUTION: msgID=153.569, attribution={accessing={identifier=com.bps.MNU, pid=63335, auid=1875722432, euid=1875722432, binary_path=/Applications/MNU.app/Contents/MacOS/MNU}, requesting={identifier=com.apple.WindowServer, pid=153, auid=88, euid=88, binary_path=/System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/Resources/WindowServer}, },
default	13:38:35.016911+0100	tccd	requestor: identifier=com.apple.WindowServer, pid=153, auid=88, euid=88, binary_path=/System/Library/PrivateFrameworks/SkyLight.framework/Versions/A/Resources/WindowServer is checking access for accessor identifier=com.bps.MNU, pid=63335, auid=1875722432, euid=1875722432, binary_path=/Applications/MNU.app/Contents/MacOS/MNU
default	13:38:35.020071+0100	MNU	Received configuration update from daemon (initial)
default	13:38:35.023250+0100	tccd	AUTHREQ_SUBJECT: msgID=153.569, subject=com.bps.MNU,
default	13:38:35.032744+0100	MNU	FRONTLOGGING: version 1
default	13:38:35.032812+0100	MNU	Registered, pid=63335 ASN=0x0,0x1d41d4
default	13:38:35.035397+0100	MNU	BringForward: pid=63335 asn=0x0-0x1d41d4 bringForward=0 foreground=0 uiElement=1 launchedByLS=1 modifiersCount=1 allDisabled=0
default	13:38:35.035475+0100	MNU	BringFrontModifier: pid=63335 asn=0x0-0x1d41d4 Modifier 0 hideAfter=0 hideOthers=0 dontMakeFrontmost=0 mouseDown=0/0 seed=0/0
default	13:38:35.036323+0100	MNU	Current system appearance, (HLTB: 2), (SLS: 1)
default	13:38:35.042552+0100	MNU	Post-registration system appearance: (HLTB: 2)
default	13:38:35.097415+0100	distnoted	register name: com.apple.xctest.FakeForceTouchDevice object: com.bps.MNU token: 1a00000013 pid: 63335
default	13:38:35.105710+0100	MNU	NSApp cache appearance:
-NSRequiresAquaSystemAppearance: 0
-appearance: (null)
-effectiveAppearance: <NSCompositeAppearance: 0x6000031c8c80
 (
    "<NSDarkAquaAppearance: 0x6000031c8c00>",
    "<NSSystemAppearance: 0x6000031dde00>"
)>
default	13:38:35.147867+0100	distnoted	register name: com.apple.nsquiet_safe_quit_give_reason object: com.bps.MNU token: 1d00000022 pid: 63335
default	13:38:35.207170+0100	MNU	Registering for test daemon availability notify post.
default	13:38:35.207370+0100	MNU	notify_get_state check indicated test daemon not ready.
default	13:38:35.210853+0100	MNU	SignalReady: pid=63335 asn=0x0-0x1d41d4
default	13:38:35.211653+0100	MNU	SIGNAL: pid=63335 asn=0x0x-0x1917396
default	13:38:35.227940+0100	tccd	-[TCCDAccessIdentity staticCode]: static code for: identifier com.bps.MNU, type: 0: 0x7f8331d6a9d0 at /Applications/MNU.app
default	13:38:35.250080+0100	MNU	Could not cast value of type '__NSCFBoolean' (0x7fff8838bb98) to 'NSString' (0x7fff883bf060).
default	13:38:35.250118+0100	kernel	AMFI: Denying core dump for pid 63335 (MNU)
default	13:38:35.250829+0100	runningboardd	[app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335] termination reported by launchd (0, 0, 6)
default	13:38:35.250869+0100	runningboardd	Removing process: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:35.251343+0100	ReportCrash	Parsing corpse data for process MNU [pid 63335]
default	13:38:35.251173+0100	runningboardd	Removing launch job for: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:35.251514+0100	runningboardd	Removed job for [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:35.251560+0100	runningboardd	Removing assertions for terminated process: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:35.258569+0100	runningboardd	Calculated state for app<application.com.bps.MNU.366350163.366350169(1875722432)>: none (role: None)
default	13:38:35.258795+0100	runningboardd	Calculated state for app<application.com.bps.MNU.366350163.366350169(1875722432)>: none (role: None)
default	13:38:35.259362+0100	launchservicesd	Hit the server for a process handle 1f48dea00000f767 that resolved to: [app<application.com.bps.MNU.366350163.366350169(1875722432)>:63335]
default	13:38:35.262023+0100	loginwindow	-[PersistentAppsSupport applicationQuit:] | for app:MNU, _appTrackingState = 2
default	13:38:35.645211+0100	ReportCrash	Saved crash report for MNU[63335] version 1.5.0 (184) to MNU_2021-03-10-133835_fme-mac002.crash
default	13:38:35.649486+0100	ReportCrash	Sending event: com.apple.stability.crash {"appVersion":"1.5.0","bundleID":"com.bps.MNU","exceptionType":13,"logwritten":1,"process":"MNU","responsibleApp":"MNU"}
default	13:38:35.649681+0100	analyticsd	Received event: com.apple.stability.crash {"appVersion":"1.5.0","bundleID":"com.bps.MNU","exceptionType":13,"logwritten":1,"process":"MNU","responsibleApp":"MNU"}
default	13:38:35.649832+0100	analyticsd	Aggregated. Transform: StabilityCrashNumerator3 Dirty: 1 Event: com.apple.stability.crash {"appVersion":"1.5.0","bundleID":"com.bps.MNU","exceptionType":13,"logwritten":1,"process":"MNU","responsibleApp":"MNU","timestamp":1615379915649414}
default	13:38:35.650000+0100	analyticsd	Aggregated. Transform: StabilityCrashNumerator3WithIncidentID Dirty: 1 Event: com.apple.stability.crash {"appVersion":"1.5.0","bundleID":"com.bps.MNU","exceptionType":13,"logwritten":1,"process":"MNU","responsibleApp":"MNU","timestamp":1615379915649414}

```

A crash file was written and this is an excerpt from it:
![image](https://user-images.githubusercontent.com/3872101/110631300-a6ec7280-81a6-11eb-8a59-ada393c3ee75.png)

**Comment by smittytone** [comment]
Came across your issue this morning, @jgoldhammer. Can't reproduce it here on two Big Sur Macs, or under Catalina. Can you run the following please — if you still have MNU installed (but not running, of course) — and let me know what values, if any, you see? Thanks!

`defaults read NSGlobalDomain > g.txt`
What is the value of AppleInterfaceStyle

`defaults read com.apple.finder`
What is the value of CreateDesktop?
What is the value of AppleShowAllFiles?

Depending on the results, I will probably post an update (1.5.2) to see if it solves your problem, or at least helps me narrow down the cause.

Tony

**Comment by jgoldhammer** [comment]
Hi Tony,

thanks for coming back to my issue. I have not installed it, but I have the extracted the values.

NSGlobalDomain
`
AppleInterfaceStyle = Dark;
`

com.apple.finder
`
 AppleShowAllFiles = 1;
`

The key CreateDesktop is not in the defaults of com.apple.finder

![image](https://user-images.githubusercontent.com/3872101/118803779-12884580-b8a4-11eb-9e4b-f4b5089fcd2b.png)

**Comment by jgoldhammer** [comment]
This error message here: 

`default 13:38:35.250080+0100 MNU Could not cast value of type '__NSCFBoolean' (0x7fff8838bb98) to 'NSString' (0x7fff883bf060).`

Regarding this error this is only a guess - maybe you try to cast the NSCFBoolean to String which does not work at all? 

See for more details https://stackoverflow.com/questions/51972952/check-for-nscfboolean-in-swift

**Comment by smittytone** [comment]
Thanks for the values, @jgoldhammer.  That helps a lot. On my systems, I see

`AppleShowAllFiles = YES;`

I'll fix the code that checks that value and post an update.

**Comment by jgoldhammer** [comment]
That fixes the problem for me now:


`defaults write com.apple.finder AppleShowAllFiles YES`

**Comment by jgoldhammer** [comment]
Thanks for investigation!

**Comment by smittytone** [comment]
Hopefully fixed in 1.5.2, out now.
